inc/extras.php

```diff
@@ -37,3 +37,54 @@ function _s_pingback_header() {
 	}
 }
 add_action( 'wp_head', '_s_pingback_header' );
+
+
+/**
+ * Adds the correct schema microdata to the body tag
+ * depending on the type of page
+ */
+function _s_schema_microdata() {
+    
+    $schema = 'http://schema.org/';
+
+    if( is_single() ) {
+        $type = "Article";
+    }
+
+    else if( is_home() || ( is_archive() && !is_author() ) ) {
+    	$type = "Blog";
+    }
+
+    else if( is_page( array('contact', 'contact-us', 'contact-page') ) ) {
+        $type = 'ContactPage';
+    }
+
+    else if( is_author() ) {
+        $type = 'ProfilePage';
+    }
+
+    else if( is_search() ) {
+        $type = 'SearchResultsPage';
+    }
+
+    else {
+        $type = 'WebPage';
+    }
+
+    echo 'itemscope="itemscope" itemtype="' . $schema . $type . '"';
+}
+
+/**
+ * Add schema microdata to thumbnails
+ */
+the_post_thumbnail('thumbnail', array('itemprop' => 'image'));
+
+
+/**
+ * Add schema microdata to nav urls
+ */
+function add_menu_atts( $atts, $item, $args ) {
+    $atts['itemprop'] = 'url';
```

Comment:
I think this is needed because the SiteNavigationElement has other properties such as anchor text.